e2e/cypress/integration/enterprise/saml/okta_login_spec.js

```diff
@@ -0,0 +1,272 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import users from '../../../fixtures/saml_users.json';
+
+//assumes that the SAML certificates+keys are already present in the config folder
+//assumes that Cypress.env.json is copied in the same folder with Cypress.json
+//assumes thet CYPRESS_* variables are set
+//In the Okta Applicaions->Okta MM App define attribute statements:
+//Name: UserType -> Value: user.userType
+//Name: IsAdmin -> Value: user.isAdmin
+//Name: IsGuest -> Value: user.isGuest
+//In the Okta Profile Editor, add following custom types:
+//- for Okta app: variablename: isAdmin(boolean, isGuest(boolean) (userType is already defined)
+//- for Okta MM app: variablename: UserType(string),IsGuest(boolean), IsAdmin(boolean)
+
+context('Okta', () => {
+    const loginButtonText = 'SAML';
+
+    const regular1 = users.regulars['samluser-1'];
+    const guest1 = users.guests['samlguest-1'];
+    const guest2 = users.guests['samlguest-2'];
+    const admin1 = users.admins['samladmin-1'];
+    const admin2 = users.admins['samladmin-2'];
+
+    const newConfig = {
+        ServiceSettings: {
+            SiteURL: Cypress.config('baseUrl'),
+        },
+        SamlSettings: {
+            Enable: true,
+            EnableSyncWithLdap: false,
+            EnableSyncWithLdapIncludeAuth: false,
+            Verify: true,
+            Encrypt: true,
+            SignRequest: true,
+            IdpUrl: Cypress.env('oktaBaseUrl') + '/app/' + Cypress.env('oktaMMAppName') + '/' + Cypress.env('oktaMMEntityId') + '/sso/saml',
+            IdpDescriptorUrl: 'http://www.okta.com/' + Cypress.env('oktaMMEntityId'),
+            IdpMetadataUrl: Cypress.env('oktaBaseUrl') + '/app/' + Cypress.env('oktaMMEntityId') + '/sso/saml/metadata',
+            AssertionConsumerServiceURL: Cypress.config('baseUrl') + '/login/sso/saml',
+            SignatureAlgorithm: 'RSAwithSHA1',
+            CanonicalAlgorithm: 'Canonical1.0',
+            IdpCertificateFile: 'saml-idp.crt',
+            PublicCertificateFile: 'saml-public.crt',
+            PrivateKeyFile: 'saml-private.key',
+            IdAttribute: '',
+            GuestAttribute: '',
+            EnableAdminAttribute: false,
+            AdminAttribute: '',
+            FirstNameAttribute: '',
+            LastNameAttribute: '',
+            EmailAttribute: 'Email',
+            UsernameAttribute: 'Username',
+            LoginButtonText: 'SAML',
+        },
+        ExperimentalSettings: {
+            UseNewSAMLLibrary: true
+        },
+        GuestAccountsSettings: {
+            Enable: true
+        },
+    };
+
+    let testSettings;
+
+    describe('SAML Login flow', () => {
+        before(() => {
+            cy.oktaAddUsers(users);
```

Comment:
Right now there are only a few tests. As the number of tests increase, it helps to reduce polluting them with prep steps and keep them focused on the actual things to test. The oktaGetOrCreateUser is there for the few cases in which we will add some temporary users (it is being called for the existing test now since it makes the test more generic/reliable) .

Reply:
Yes, but we are unnecessarily adding a lot of users & then deleting them at the end without any usage. We can add this step in the future if all the users are used across.

Reply:
Yeah, I agree. Can we have couple of regular users only? It takes some time to create and assign all users. Also, sometimes I'm seeing 404 error when assigning user to application on retesting multiple times.

Reply:
We can reduce the set of test to the ones that we are actually using in the current tests right now (regular1, admin1+2, guest1+2) and increase the set as we add more tests then. Would that work?

Reply:
Yup, I'm good with it.